integration/lsp.test.ts

```diff
@@ -0,0 +1,125 @@
+function make_request(data) {
+    data.jsonrpc = "2.0";
+    data.id = 1;
+    const json = JSON.stringify(data);
+    return `Content-Length: ${json.length}\r\n\r\n${json}`;
+}
+
+function parse_response(data) {
+    let json = JSON.parse(data.split('\r\n')[2]);
+    delete json.id;
+    delete json.jsonrpc;
+    return json;
+}
+
+test('initialize', async () => {
+    import('@influxdata/flux-lsp-node')
+    .then(async ({Server}) => {
+        const expected = {
+            "result": {
+                "capabilities": {
+                    "textDocumentSync":1,
+                    "referencesProvider":true,
+                    "definitionProvider":true,
+                    "renameProvider":true,
+                    "foldingRangeProvider":true,
+                    "documentSymbolProvider":true,
+                    "documentFormattingProvider":true,
+                    "completionProvider": {
+                        "resolveProvider":true,
+                        "triggerCharacters": [".",":","(",",","\""]
+                    },
+                    "signatureHelpProvider": {
+                        "triggerCharacters": ["("],
+                        "retriggerCharacters": ["("]
+                    },
+                    "hoverProvider":true
+                }
+            }
+        };
+
+        const server = new Server(false, false);
+        const request = make_request({method: "initialize"});
+
+        const response = await server.process(request);
+        const error = response.get_error();
+        expect(error).toBe(undefined);
+        const message = parse_response(response.get_message());
+        expect(message).toStrictEqual(expected);
+    });
+});
+
+test('shutdown', () => {
+    import('@influxdata/flux-lsp-node')
+    .then(async ({Server}) => {
+        const expected = {"result": null };
+        const server = new Server(false, false);
+        const request = make_request({method: "shutdown"});
+
+        const response = await server.process(request);
+        const error = response.get_error();
+        expect(error).toBe(undefined);
+        const message = parse_response(response.get_message());
+        expect(message).toStrictEqual(expected);
+    });
+});
+
+test('textDocument/formatting', () => {
+    import('@influxdata/flux-lsp-node')
```

Comment:
My understanding is that this would import the version published on NPM. Is that what we want, or do we want to test the version of the package that's being committed/merged? If this is running in CI, I think we want the latter.

Reply:
In `packages.json`, we're installing our locally built package manually in the `install-pkg` step, and never referencing it otherwise, i.e. in `dependencies`. I've verified that this does run against the locally built wasm, as I had to debug it originally.

Reply:
Ah, I see. Thanks for the clarification.